### ISSUE TITLE
Evaluate removing sigsetjmp() block in task_tracker.c

### ISSUE THREAD

**Issue by sumedhpathak** [citusdata/citus#296]
We're refactoring the task tracker process as a background worker process. As part of that thread, we discussed removing the sigsetjmp() block in task_tracker.c. I'm including related history on GitHub extensify-11: https://github.com/citusdata/citusdb-extensify/pull/7

> > I've read more of the related code and I think this would be better handled as a separate task. Currently the sigsetjmp() logic is important because of this block:

``` /*
     * We run validation and cache cleanup functions as this process is starting
     * up. If these functions throw an error, we won't try running them again.
     */
    if (processStartUp)
    {
        processStartUp = false;

        /*
         * The task tracker process requires the client executor library to be
         * loaded in memory. This loading is done during postmaster and backend
         * start up; and we simply check here that all symbols are loaded.
         */
        ValidateLibraryLoad();

        /* clean up old files in the job cache */
        TrackerCleanupJobDirectories();

        /* clean up schemas in the job cache */
        TrackerCleanupJobSchemas();
    }
```

This makes sense. I'll open another task in Trello, and also update two existing tasks with the following:
1. We introduced ValidateLibraryLoad() to make sure that the libpqclientexecutor shared library is loaded. Once we remove this additional shared object, we may just remove this validation function. I'll update the corresponding Trello item to not lose track of this.
2. TrackerCleanupJobDirectories() doesn't use locks. I'm guessing we wrap this function in PG_TRY/PG_CATCH blocks.
3. TrackerCleanupJobSchemas() currently doesn't do anything. We have an open Trello item on refactoring this function. If we decide to remove this functionality or have schemas automatically cleaned up on shut down, we don't need this function / locks around it. I'll also note this dependency in Trello tasks.

In summary, ValidateLibraryLoad() and TrackerCleanupJobSchemas() relate to Trello items 6 and 7. If we decide to remove these functions, we can also remove the sigsetjmp() block.

One question in my mind is how risky it is to encapsulate lock allocation inside PG_TRY/PG_CATCH blocks.
